bridges/snowbridge/pallets/system-frontend/src/lib.rs

```diff
@@ -0,0 +1,215 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: 2023 Snowfork <hello@snowfork.com>
+//! Frontend which will be deployed on AssetHub for calling the V2 system pallet
+//! on BridgeHub.
+//!
+//! # Extrinsics
+//!
+//! * [`Call::create_agent`]: Create agent for any kind of sovereign location on Polkadot network.
+//! * [`Call::register_token`]: Register Polkadot native asset as a wrapped ERC20 token on Ethereum.
+#![cfg_attr(not(feature = "std"), no_std)]
+#[cfg(test)]
+mod mock;
+
+#[cfg(test)]
+mod tests;
+
+#[cfg(feature = "runtime-benchmarks")]
+mod benchmarking;
+
+pub mod weights;
+pub use weights::*;
+
+use frame_support::{pallet_prelude::*, traits::EnsureOrigin};
+use frame_system::pallet_prelude::*;
+use sp_std::prelude::*;
+use xcm::prelude::*;
+use xcm_executor::traits::TransactAsset;
+
+#[cfg(feature = "runtime-benchmarks")]
+use frame_support::traits::OriginTrait;
+
+pub use pallet::*;
+use snowbridge_core::AssetMetadata;
+
+#[derive(Encode, Decode, Debug, PartialEq, Clone, TypeInfo)]
+pub enum ControlCall {
+	#[codec(index = 1)]
+	CreateAgent { location: Box<VersionedLocation>, fee: u128 },
+	#[codec(index = 2)]
+	RegisterToken { asset_id: Box<VersionedLocation>, metadata: AssetMetadata, fee: u128 },
+}
+
+#[allow(clippy::large_enum_variant)]
+#[derive(Encode, Decode, Debug, PartialEq, Clone, TypeInfo)]
+pub enum SnowbridgeControl {
+	#[codec(index = 90)]
+	Control(ControlCall),
+}
```

Comment:
@yrong 

1. Move these to the outbound-queue primitives crate
2. Rename as follows:

```suggestion
#[derive(Encode, Decode, Debug, PartialEq, Clone, TypeInfo)]
pub enum EthereumSystemCall {
	#[codec(index = 1)]
	CreateAgent { location: Box<VersionedLocation>, fee: u128 },
	#[codec(index = 2)]
	RegisterToken { asset_id: Box<VersionedLocation>, metadata: AssetMetadata, fee: u128 },
}

#[allow(clippy::large_enum_variant)]
#[derive(Encode, Decode, Debug, PartialEq, Clone, TypeInfo)]
pub enum BridgeHubRuntime {
	#[codec(index = 90)]
	EthereumSystem(EthereumSystemCall),
}
```

Reply:
Also the index of the `EthereumSystem` pallet on BH could differ between BH runtimes, or are we relying on the convention that they are the same? Please verify.

Reply:
It's the index of `EthereumSystemV2`. Yeah, we can make it same. 

Or can differ with multiple features if not the same, but I don't think it's necessary.

Reply:
Rename in https://github.com/vgeddes/polkadot-sdk/pull/6/commits/8d32278638aa1b28e9e3304b95b79e8327d92ed0

Just curious why you'd prefer to move to the primitives crate, I'd assume it's only used by the caller on AH?